claat/parser/md/parse.go

```diff
@@ -849,6 +849,10 @@ func link(ds *docState) types.Node {
 func text(ds *docState) types.Node {
 	bold := isBold(ds.cur)
 	italic := isItalic(ds.cur)
+	if isBoldAndItalic(ds.cur) {
```

Comment:
Add a comment explaining why we need this a separate case from bold and italic individually.